tcrc-35/vnedor filter api hook up

holahoon opened the pull request:
[Docs regarding query-string](https://github.com/sindresorhus/query-string)

Solves [TCRC-35](https://opuslogica.atlassian.net/browse/TCRC-35)

- Live filter on desktop
- Button filter on mobile

nickfarina commented:
I noticed some sort of bug that occured after selecting a bunch of filters and then reloading the page. for reference, the url I refreshed was http://localhost:3000/alternate-spd?ages=item1|item3&counties=item2&page=1&serviceCategories=item2

![Screen Shot 2024-04-23 at 11 28 56 PM](https://github.com/opuslogica/tcrc-spd/assets/5157728/00ce2f86-e094-4c71-9340-415c51e51f6a)

holahoon commented:
There was a bug with the url query params. With the query-string function I converted an an object with array values to be stringified and separated by `|`, so it'd be `{key: [1, 2]}` -> `key=1|2&...`. The problem was if the array has only one element, it wouldn't be stringified and separated by `|`, but rather it just becomes `key=1`. Then when this gets parsed, it is not converted into an array, but a string. Since multiselect component expects an array of strings, it complains because the selected value is of type string 🤦🏻
I changed the separator type to be "colon-list-separator".
So it'd be like:
```javascript
import queryString from 'query-string';

queryString.stringify({foo: ['one', 'two']}, {arrayFormat: 'colon-list-separator'});
//=> 'foo:list=one&foo:list=two'
```
